themes/default/content/challenge/startup-in-a-box.md

```diff
@@ -508,9 +513,410 @@ expect(await page.title()).toBe("Pulumi Challenge");
+const httpForwardingRule = new gcp.compute.GlobalForwardingRule("http-forwarding-rule", {
+    ipAddress: ip.address,
+    ipProtocol: "TCP",
+    portRange: "80",
+    target: httpProxy.selfLink,
+});
+```
+
+#### Step 5. Introducing ComponentResources
+
+Now… we can continue to add resource after resource, but Pulumi is more than that. We can build our own reusable components. Let’s refactor what we have above into a CdnWebsite component at ```pulumi-challenge/cdn-website.ts```
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as gcp from "@pulumi/gcp";
+import * as synced_folder from "@pulumi/synced-folder";
+
+export class CdnWebsite extends pulumi.ComponentResource {
+    private bucket: gcp.storage.Bucket;
+    private backendBucket: gcp.compute.BackendBucket;
+    private ip: gcp.compute.GlobalAddress;
+    private httpForwardingRule: gcp.compute.GlobalForwardingRule;
+
+    constructor(name: string, args: any, opts?: pulumi.ComponentResourceOptions) {
+        super("pulumi:challenge:CdnWebsite", name, args, opts);
+
+        this.bucket = new gcp.storage.Bucket("mybucket", {
+            location: "US"
+        });
+
+        const bucketIamBinding = new gcp.storage.BucketIAMBinding("bucket-iam-binding", {
+            bucket: this.bucket.name,
+            role: "roles/storage.objectViewer",
+            members: ["allUsers"],
+        });
+
+        const config = new pulumi.Config();
+        const path = config.get("path") || "./website";
+
+        const syncedFolder = new synced_folder.GoogleCloudFolder("synced-folder", {
+            path: path,
+            bucketName: this.bucket.name,
+        });
+
+        // Google Cloud Load Balancer Backend
+        this.backendBucket = new gcp.compute.BackendBucket("backend-bucket", {
+            bucketName: this.bucket.name,
+            enableCdn: true,
+        });
+
+        // CDN Configuration
+        this.ip = new gcp.compute.GlobalAddress("ip", {});
+        const urlMap = new gcp.compute.URLMap("url-map", { defaultService: this.backendBucket.selfLink });
+        const httpProxy = new gcp.compute.TargetHttpProxy("http-proxy", { urlMap: urlMap.selfLink });
+
+        this.httpForwardingRule = new gcp.compute.GlobalForwardingRule("http-forwarding-rule", {
+            ipAddress: this.ip.address,
+            ipProtocol: "TCP",
+            portRange: "80",
+            target: httpProxy.selfLink,
+        });
+
+        // We also need to register all the expected outputs for this
+        // component resource that will get returned by default.
+        this.registerOutputs({
+            bucketName: this.bucket.id,
+            cdnUrl: pulumi.interpolate`http://${this.ip.address}`
+        });
+    }
+
+    get url(): pulumi.Output<string> {
+        return pulumi.interpolate`http://${this.ip.address}`;
+    }
+}
+```
+
+Now we can consume this! Awesome. Back in `pulumi-challenge/index.ts`, we now have this:
```

Comment:
is this the first time I update the index.ts file. So, it is to `replace the index.ts with the following code`?